docs/topics/forms/formsets.txt

```diff
@@ -48,13 +48,14 @@ following example will create a formset class to display two blank forms:
 
     >>> ArticleFormSet = formset_factory(ArticleForm, extra=2)
 
-Iterating over a formset will render the forms in the order they were
-created. You can change this order by providing an alternate implementation for
-the ``__iter__()`` method.
+Iterating over a formset will render the forms in the order they were created.
+The order can be adjusted by changing how the formset is iterable_.
```

Comment:
This has made me wonder if this is the best advice.

Another option is something like this:
```
The order can be adjusted by overriding the ``forms``` property::

    from django.forms import BaseFormSet
    from django.utils.functional import cached_property


    class BaseArticleFormSet(BaseFormSet):
        @cached_property
        def forms(self):
            forms = super().forms
            return list(reversed(forms))
```

The whole, "you can override the iter" was added as part of ticket-14655 in 7adffaeaf6dfa22db0b6b2a29632b9150c7ac732 which was quite a long time ago. I'm still tempted that we remove it and folks figure it out themselves

Reply:
Took another look at the formset docs, and I see your point now 👍🏾. 

Earlier I was tunneled on the idea that overriding iter etc was the way to reorder forms (having got that idea from the docs), and I thought there were good use cases, and that we needed to keep some form of documentation on it.

But I've realised users can input initial data in any order they like, so they can control the order at that stage if needed. Also, reordering doesn't really make sense when there’s no initial data, and even if initially empty forms are later filled in, I still can’t think of a good reason for reordering them

Reply:
@nessita want to highlight this previous comment/discussion

Reply:
I'd like to point out that the reporter initially lacked experience with Python’s iterable and sequence protocols when they created the report, but they figured it out soon after creating the ticket, suggesting that suggests they found guidance sufficient for their needs. (But perhaps the ticket might not have existed if we'd linked to the guidance on the protocols before.)

If we acknowledge the need for reordering formsets after creation, then mentioning the iterable and sequence protocols alone might suffice for users who are already familiar with them. Those unfamiliar could follow the links to Python’s documentation. However, if our goal is to guide users specifically on reordering formsets, then it would be more helpful to explicitly mention how overriding these protocols achieves that. ("The order can be changed by..." or something like that) Otherwise, users looking to reorder might not realise this guidance meets their needs.

OTOH if we decide to “wontfix” this ticket, we still have to recognize that mentioning __iter__ and __getitem__ without linking to documentation isn’t ideal. A link to Python’s docs would improve accessibility for users.

And if we choose to remove the reordering advice altogether as an edge case, then IMO referencing these protocols may be unnecessary.